### PR TITLE
Updates to approve device on desktop

### PR DESCRIPTION
**Pull request by atavism**
Depends on https://github.com/getlantern/flashlight/pull/1369

Resolves https://github.com/getlantern/engineering/issues/1298

### CODE REVIEW

File: lib/common/session_model.dart [hunk 1]
```diff
@@ -331,16 +331,22 @@ class SessionModel extends Model {
...
-    return methodChannel.invokeMethod('approveDevice', <String, dynamic>{
-      'code': code,
-    }).then((value) => value as String);
+  Future<String> approveDevice(String code) async {
+    if (isMobile()) {
+      return methodChannel.invokeMethod('approveDevice', <String, dynamic>{
+        'code': code,
+      }).then((value) => value as String);
+    }
+    return await ffiApproveDevice(code.toNativeUtf8());
```
Comment (jigar-f) [inline lib/common/session_model.dart:340]: @atavism Can we use compute here to avoid UI freeze?

File: lib/common/session_model.dart [hunk 2]
```diff
@@ -331,16 +331,22 @@ class SessionModel extends Model {
...
-    return methodChannel.invokeMethod('removeDevice', <String, dynamic>{
-      'deviceId': deviceId,
-    });
+  Future<void> removeDevice(String deviceId) async {
+    if (isMobile()) {
+      return methodChannel.invokeMethod('removeDevice', <String, dynamic>{
+        'deviceId': deviceId,
+      });
+    }
+    return await ffiRemoveDevice(deviceId.toNativeUtf8());
```
Comment (jigar-f) [inline lib/common/session_model.dart:349]: @atavism Can we use compute here to avoid UI freeze?

File: lib/ffi.dart [hunk 3]
```diff
@@ -40,6 +40,29 @@ Future<User> ffiUserData() async {
...
+  checkAPIError(result, 'wrong_device_linking_code'.i18n);
+  // refresh user data after successfully linking device
+  await ffiUserData();
+  return json;
+}
+
+Future<void> ffiRemoveDevice(deviceId) async {
+  final json = await _bindings.removeDevice(deviceId).cast<Utf8>().toDartString();
+  final result = LinkResponse.create()..mergeFromProto3Json(jsonDecode(json));
+  checkAPIError(result, 'cannot_remove_device'.i18n);
```
Comment (jigar-f) [inline lib/ffi.dart:62]: @atavism After removing, I think we need to update device list as well?

**Review by jigar-f — APPROVED** [review]
@atavism Just a couple of small comments, otherwise LGTM!

File: lib/common/session_model.dart (hunk 1 above)
Comment (atavism) [reply lib/common/session_model.dart:340]: Done.. thanks!

File: lib/ffi.dart (hunk 3 above)
Comment (atavism) [reply lib/ffi.dart:68]: Done, thanks!

**Comment by atavism** [comment]
> @atavism Just a couple of small comments, otherwise LGTM!

Thanks for the review! I'm not sure why your own device isn't showing up (I think it may be unrelated to the changes here). 

Can you check backoffice to see if the device shows up when you look up your user?

**Pull request merged**